src/config.rs

```diff
@@ -108,6 +114,103 @@ impl Config {
     pub fn demo_crates(&self) -> &DemoCrates {
         &self.demo_crates
     }
+
+    pub fn check() {
+        let result = Self::check_all();
+
+        if let Err(e) = result {
+            use std::process::exit;
+            error!("check-config failed: {}", e);
+            exit(1);
+        }
+    }
+
+    fn check_all() -> Result<()> {
+        // A simplified config definition, to validate it for duplicate keys.
+        #[derive(Deserialize)]
+        struct Config {
+            #[serde(rename = "crates")]
+            _crates: ::toml::Value,
+            #[serde(rename = "github-repos")]
+            _github_repos: ::toml::Value,
+        }
+
+        let buffer = Self::load_as_string()?;
+        let _cfg: Config = ::toml::from_str(&buffer)?;
```

Comment:
Can you just serialize the whole configuration into a `::toml::Value`? It should have the same effect.